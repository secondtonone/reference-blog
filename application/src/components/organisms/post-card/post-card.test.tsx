import renderer from 'react-test-renderer';
import PostCard from './post-card';
import fixtureArticles from '../../../__fixtures__/articles';

jest.mock('~/components/containers/svg-icon', () => ({ code }) => <span>{code}</span>);

describe('Post card works', () => {
  it('vertical with image', () => {
    const content = fixtureArticles[0];

    const tree = renderer.create(<PostCard {...content} />);

    expect(tree).toMatchSnapshot();
  });
});
