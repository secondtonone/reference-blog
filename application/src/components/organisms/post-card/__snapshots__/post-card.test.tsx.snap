// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post card works vertical with image 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 uifxx"
  data-test="post-card-item-en"
>
  <div
    className="box-adaptive__BoxStyled-sc-g2wy39-0 VYInM post-card-image__ImageStyled-sc-r39eru-0 fHpPev"
    data-test="image"
  >
    <a
      data-test="image-link"
      href="/white-hat-seo"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <img
        alt="Article Image"
        className="image__ImageStyled-sc-rj8okk-0 hueYwk lazyload"
        data-src="https://storage.googleapis.com/blog-rc/testing/article-list/1.jpg"
        height="190"
        src={null}
        width="158"
      />
    </a>
  </div>
  <article
    className="post-card-content__ContentStyled-sc-gda4z3-1 cBpLBb"
    data-test="content-en"
  >
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 jDLZGW"
    >
      <div
        className="post-card-category__DetailsStyled-sc-yr5537-2 bhQdVr"
      >
        <p
          className="text-content__TextStyled-sc-1stkrsz-0 kszZkd post-card-category__CategoryStyled-sc-yr5537-3 lnUjcU"
          data-test="category"
        >
          <a
            href="/blog/category/advertising"
            onClick={[Function]}
            onMouseEnter={[Function]}
            target="_blank"
          >
            Advertising
          </a>
        </p>
        <div
          className="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq post-card-category__TimeToReadStyled-sc-yr5537-0 fZVgVc"
        >
          <span
            className="text-content__TextStyled-sc-1stkrsz-0 eljVVb"
            data-test="time-to-read"
          >
            15
             
            min read
          </span>
        </div>
      </div>
    </div>
    <div
      className="box-adaptive__BoxStyled-sc-g2wy39-0 csiPyy"
    >
      <h3
        className="text-content__TextStyled-sc-1stkrsz-0 hHiLZj"
        data-test="text-content"
        style={
          Object {
            "wordBreak": "break-word",
          }
        }
      >
        <a
          className="post-card-content__HeadingLinkStyled-sc-gda4z3-0 iOoMty"
          data-test="title"
          href="/white-hat-seo"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          The Ultimate Guide to Google Search Operators and Google Search Commands
        </a>
      </h3>
      <span
        className="text-content__TextStyled-sc-1stkrsz-0 kdmpti post-card-content__CaptionStyled-sc-gda4z3-3 iLbRjP"
        data-test="description"
      >
        In this guide, weâ€™ll delve deep into the basics of search operators and search commands, helping you understand how to wield these powerful tools effectively and move on to advanced commands and operators.
      </span>
      <div
        className="box-adaptive__BoxStyled-sc-g2wy39-0 cGxrni post-card-content__MetaStyled-sc-gda4z3-4 cgjjhY"
        data-test="meta"
      >
        <a
          data-test="author-name"
          href="/user/olga-andrienko"
          onClick={[Function]}
          onMouseEnter={[Function]}
          target="_blank"
        >
          <span
            className="text-content__TextStyled-sc-1stkrsz-0 izRvWD post-card-content__NoWrapStyled-sc-gda4z3-2 jqrfVT"
            data-test="text-content"
          >
            Olga Andrienko
          </span>
        </a>
        <span
          className="text-content__TextStyled-sc-1stkrsz-0 kHbJxs post-card-content__NoWrapStyled-sc-gda4z3-2 jqrfVT"
          data-test="date"
        >
          Aug 20, 2020
        </span>
      </div>
    </div>
  </article>
</div>
`;
